codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -56,22 +58,33 @@ public DefaultHttp2LocalFlowController(Http2Connection connection,
         this.frameWriter = checkNotNull(frameWriter, "frameWriter");
         windowUpdateRatio(windowUpdateRatio);
 
-        // Add a flow state for the connection.
-        final Http2Stream connectionStream = connection.connectionStream();
-        connectionStream.setProperty(FlowState.class, new FlowState(connectionStream, initialWindowSize));
+        connection.connectionStream().localFlowState(
+                new DefaultFlowState(connection.connectionStream(), initialWindowSize));
 
         // Register for notification of new streams.
         connection.addListener(new Http2ConnectionAdapter() {
             @Override
             public void onStreamAdded(Http2Stream stream) {
-                stream.setProperty(FlowState.class, new FlowState(stream, 0));
+                stream.localFlowState(REDUCED_FLOW_STATE);
             }
 
             @Override
             public void onStreamActive(Http2Stream stream) {
                 // Need to be sure the stream's initial window is adjusted for SETTINGS
                 // frames which may have been exchanged while it was in IDLE
-                state(stream).window(initialWindowSize);
+                stream.localFlowState(new DefaultFlowState(stream, initialWindowSize));
+            }
+
+            @Override
+            public void onStreamClosed(Http2Stream stream) {
```

Comment:
@Scottmitch I think when a stream is closed, we should automatically consume all of it's remaining unconsumed bytes so that they will be given back to the connection window.  The `REDUCED_FLOW_STATE`'s `consumeBytes` can just be a no-op.  WDYT?